### PR TITLE
Divyanshu | Test |  Customer/Vendor wise MRP and Discount on variant

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/85ztz8twr

### CODE REVIEW

File: apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.scss [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        .icon-search-icon {
+            position: absolute;
+            left: 0;
+            right: 0;
+            width: 4.2rem;
+            text-align: center;
+            color: var(--color-abbey);
+        }
+    }
+    .vR {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.scss]: don't use capital letter in css. Change it to .v-r or something else. Fix this issue in all places.

File: apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.ts [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    /* This will hold local JSON data */
+    public localeData: any = {};
+    /* This will hold common JSON data */
+    public commonLocaleData: any = {};
+
+
+    constructor(
+        private renderer: Renderer2,
+        private zone: NgZone,
+        private _inventoryService: InventoryService,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.ts]: don't use underscore

File: apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.ts [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+        }
+    }
+
+    /**
+     * Lifecycle hook for destroy component
+     *
+     * @memberof AdvanceListItemsPopupComponent
+     */
+    public ngOnDestroy(): void {
+        document.querySelector("body")?.classList?.remove("cmd-k");
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.ts]: don't use cmd-k class name here

File: apps/web-giddh/src/app/new-inventory/component/custom-price/customer-wise/customer-wise.component.html [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    </div>
+    <div class="custom-grid-wrapper" *ngIf="!isLoading">
+        <div class="mat-list-wrap">
+            <div class="horizontal-scroll" id="horizontal-scroll">
+                <div class="inventory-sideabar border-grid">
+                    <div class="inventory-sub-sideabar">
+                        <mat-list class="item-list border-grid pt-0">
+                            <div class="list-head-gap">
+                                <div class="search-list position-relative mr-t15 mr-b15" [formGroup]="searchForm">
+                                    <text-field type="text" [placeholder]="'Search'" formControlName="userSearch"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/custom-price/customer-wise/customer-wise.component.html]: use translation. Fix this issue everywhere

File: apps/web-giddh/src/app/new-inventory/component/custom-price/customer-wise/customer-wise.component.ts [hunk 5]
```diff
@@ -0,0 +1,938 @@
...
+     * @private
+     * @param {*} variant
+     * @return {*}  {UntypedFormGroup}
+     * @memberof CustomerWiseComponent
+     */
+    private initVariantForm(variant: any): UntypedFormGroup {
+        return this.formBuilder.group({
+            price: [variant?.price, Validators.required],
+            quantity: [variant?.defaultQuantity, Validators.required],
+            discountExclusive: [variant?.discountExclusive ?? false, Validators.required], // to ask rishi ji
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/custom-price/customer-wise/customer-wise.component.ts]: remove the comment

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 6]
```diff
@@ -1477,16 +1478,107 @@ export class InventoryService {
...
+     * @memberof InventoryService
+     */
+    public getAllDiscount(request: any): Observable<BaseResponse<any, any>> {
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        let apiUrl = this.config.apiUrl + INVENTORY_API.GET_ALL_DISCOUNTS?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))?.replace(':page', encodeURIComponent(request.page))?.replace(':count', encodeURIComponent(request.count))?.replace(':uniqueName', encodeURIComponent(request.uniqueName))?.replace(':query', encodeURIComponent(request.query || ''));
+
+        return this.http.get(apiUrl);
+    }
+
+    public deleteDiscountRecord(request: any): Observable<BaseResponse<any, any>> {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/inventory.service.ts:1559]: add documentation on all methods you added

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.scss (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.scss]: Done

File: apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.ts (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/new-inventory/component/custom-price/advance-list-items-popup/advance-list-items-popup.component.ts]: Removed not in use

File: apps/web-giddh/src/app/new-inventory/component/custom-price/customer-wise/customer-wise.component.html (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/new-inventory/component/custom-price/customer-wise/customer-wise.component.html]: done

File: apps/web-giddh/src/app/services/inventory.service.ts (hunk 6 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/services/inventory.service.ts:1559]: done

File: apps/web-giddh/src/app/new-inventory/component/custom-price/customer-wise/customer-wise.component.ts (hunk 5 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/new-inventory/component/custom-price/customer-wise/customer-wise.component.ts]: done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**